unity-renderer/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/LoadWrapper/Tests/LoadWrapperShould.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections;
+using DCL;
+using DCL.Components;
+using DCL.Helpers;
+using DCL.Models;
+using NSubstitute;
+using Tests;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Assert = UnityEngine.Assertions.Assert;
+using Object = UnityEngine.Object;
+
+public class LoadWrapperShould : IntegrationTestSuite
+{
+    protected override PlatformContext CreatePlatformContext() { return DCL.Tests.PlatformContextFactory.CreateWithGenericMocks(WebRequestController.Create()); }
+
+    [UnityTest]
+    public IEnumerator UnloadWhenEntityDestroyedBeforeFinishing()
+    {
+        GameObject meshRootGameObject = new GameObject();
+
+        string url = TestAssetsUtils.GetPath() + "/GLB/Lantern/Lantern.glb";
+
+        IDCLEntity entity = Substitute.For<IDCLEntity>();
+        entity.meshRootGameObject.Returns(meshRootGameObject);
+
+        LoadWrapper_GLFT_Overload wrapper = new LoadWrapper_GLFT_Overload();
+        wrapper.entity = entity;
+        wrapper.customContentProvider = new ContentProvider();
+
+        bool loaded = false;
+        bool failed = false;
+        bool unloaded = false;
+
+        wrapper.OnUnload += () => unloaded = true;
+
+        wrapper.Load(url, loadWrapper => loaded = true, loadWrapper => failed = true );
+
+        entity.OnCleanupEvent?.Invoke(entity);
+
+        yield return new WaitUntil(() => loaded || failed || unloaded);
+
+        Object.Destroy(meshRootGameObject);
+
+        Assert.IsTrue(unloaded, "Unload should be called if entity is cleaned up while loading mesh");
+    }
+
+    class LoadWrapper_GLFT_Overload : LoadWrapper_GLTF
```

Comment:
```suggestion
    class LoadWrapper_GLTF_Overload : LoadWrapper_GLTF
```

Reply:
I think we can use `NSubstitute.ForPartsOf()` and `Unload().Received()` instead of inheriting like this.

Reply:
Will `NSubstitute` use the original `Unload` method or it will mock it?
I'm not entirely sure about using `NSubstitute` for the class we are actually testing.
For example, if `NSubstitute` mocks the `Unload` we will never know if there is an Exception happening inside that method under this conditions.
I will check that out, but if the original method is not called I think we should keep that as it is :)

Reply:
Yep, `NSubstitute.ForPartsOf<>` will use the implementation unless you override it. Brian's approach should be correct here.